### PR TITLE
Link Field

### PR DESCRIPTION
**Pull request by agriffard**
See #352 
Link field implementation.

### CODE REVIEW

File: src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Drivers/LinkFieldDriver.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+using Orchard.ContentManagement.Metadata.Models;
+using Orchard.DisplayManagement.ModelBinding;
+using Orchard.DisplayManagement.Views;
+using Orchard.Localization;
+
+namespace Orchard.ContentFields.Fields
+{
+    public class LinkFieldDisplayDriver : ContentFieldDisplayDriver<LinkField>
+    {
+        public LinkFieldDisplayDriver()
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Drivers/LinkFieldDriver.cs]: Resolve `ILocalizer<LikFieldDisplayDriver>` and use it.

File: src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Drivers/LinkFieldDriver.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        public override IDisplayResult Display(LinkField field, BuildFieldDisplayContext context)
+        {
+            return Shape<DisplayLinkFieldViewModel>("LinkField", model =>
+            {
+                model.Field = field;
+                model.Part = context.ContentPart;
+                model.PartFieldDefinition = context.PartFieldDefinition;
+            })
+            .Location("Content");
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Drivers/LinkFieldDriver.cs]: Add `.Location("SummaryAdmin", "");` To hide in the summary admin.

File: src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Drivers/LinkFieldDriver.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        private static string GetPrefix(UpdateFieldEditorContext context)
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Drivers/LinkFieldDriver.cs]: Not necessary. The base class creates one like this already.

**Comment by sebastienros** [comment]
Have you seen this? https://www.drupal.org/docs/8/core/modules/link/overview

We should remove the target and let the display mode decide how it should behave. We can just generate a simple link for the default display. It feels too much to let an editor specify this. Same for `rel="nofollow"`.

**Comment by agriffard** [comment]
OK, changes done to LinkField.

File: src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Indexing/LinkFieldIndexHandler.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+using Orchard.Indexing;
+
+namespace Orchard.ContentFields.Indexing
+{
+    public class LinkFieldIndexHandler : ContentFieldIndexHandler<LinkField>
+    {
+        public override Task BuildIndexAsync(LinkField field, BuildFieldIndexContext context)
+        {
+            var options = context.Settings.ToOptions();
+            context.DocumentIndex.Entries.Add(context.Key, new DocumentIndex.DocumentIndexEntry(field.Value, DocumentIndex.Types.Text, options));
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Indexing/LinkFieldIndexHandler.cs]: A link should never be analyzed. And should probably be stored by default.

File: src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Settings/LinkFieldSettingsDriver.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                .Location("Content");
+        }
+
+        public override async Task<IDisplayResult> UpdateAsync(ContentPartFieldDefinition partFieldDefinition, UpdatePartFieldEditorContext context)
+        {
+            var model = new LinkFieldSettings();
+
+            await context.Updater.TryUpdateModelAsync(model, Prefix);
+
+            context.Builder.Hint(model.Hint);
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Settings/LinkFieldSettingsDriver.cs]: `context.Builder.MergeSettings(model);`

File: src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Fields/LinkField.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿using Orchard.ContentManagement;
+
+namespace Orchard.ContentFields.Fields
+{
+    public class LinkField : ContentField
+    {
+        public string Value { get; set; }
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Fields/LinkField.cs]: Shouldn't we call it `Url` ?

File: src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Indexing/LinkFieldIndexHandler.cs (hunk 4 above)
Comment (agriffard) [inline src/Orchard.Cms.Web/Modules/Orchard.ContentFields/Indexing/LinkFieldIndexHandler.cs]: Do I have anything to change on the LinkFieldIndexHandler?

**Pull request merged**